cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -150,21 +152,39 @@ func (o *ScaleUpOrchestrator) ScaleUp(
 			skippedNodeGroups[nodeGroup.Id()] = NotReadyReason
 			continue
 		}
-
 		if skipReason := o.IsNodeGroupResourceExceeded(resourcesLeft, nodeGroup, nodeInfo); skipReason != nil {
 			skippedNodeGroups[nodeGroup.Id()] = skipReason
 			continue
 		}
 
-		option := o.ComputeExpansionOption(podEquivalenceGroups, nodeGroup, nodeInfo, upcomingNodes)
-		if len(option.Pods) > 0 && option.NodeCount > 0 {
-			expansionOptions[nodeGroup.Id()] = option
-		} else {
+		validNodeGroups = append(validNodeGroups, nodeGroup)
+	}
+
+	// Calculate expansion options
+	schedulablePods := map[string][]*apiv1.Pod{}
+	var options []expander.Option
+
+	for _, nodeGroup := range validNodeGroups {
+		schedulablePods[nodeGroup.Id()] = o.SchedulablePods(podEquivalenceGroups, nodeGroup, nodeInfos[nodeGroup.Id()])
+	}
+
+	for _, nodeGroup := range validNodeGroups {
+		option := o.ComputeExpansionOption(schedulablePods[nodeGroup.Id()], nodeGroup, nodeInfos[nodeGroup.Id()], upcomingNodes)
+		if len(option.Pods) == 0 || option.NodeCount == 0 {
 			klog.V(4).Infof("No pod can fit to %s", nodeGroup.Id())
+			continue
+		}
+
+		// Compute similar node groups
```

Comment:
Sounds good, moved the `ComputeSimilarNodeGroups` to `ComputeExpansionOption`. Similar node groups need to be recomputed after node group creation, so that makes sense to keep it separate.